src/daemon/rpmostreed-transaction-types.c

```diff
@@ -1054,6 +1057,12 @@ initramfs_state_transaction_execute (RpmostreedTransaction *transaction,
   rpmostree_origin_set_regenerate_initramfs (origin, self->regenerate, self->args);
   rpmostree_sysroot_upgrader_set_origin (upgrader, origin);
 
+  RpmOstreeSysrootUpgraderLayeringType layering_type;
+  gboolean layering_changed = FALSE;
+  if (!rpmostree_sysroot_upgrader_prep_layering (upgrader, &layering_type, &layering_changed,
```

Comment:
Maybe let's simplify this flow by just automatically calling `prep_layering` in `deploy` if `self->layering_initialized == FALSE` and drop the related `g_assert`?